### ISSUE TITLE
Search: Polish the new search result filter panel 

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#59385]
In this PR https://github.com/sourcegraph/sourcegraph/pull/59178 we added the new filter panel UI but there were some leftovers, in this issue we should fix most of them, here is a short list of important things

- Filtering the language list doesn't seem to work (it ignores case register)  
- Replace "Show all" with "Show more [X]" in the truncated filter list UI
- Use only repository name in the repo filter panel (ignore host part of URL)
- Add `count:all` filter if results got their limit 
- Make search filter input with standard size (not small)
- Repository icons should be outlined 
- Repo names should have a hover with the full name
- The numbers should respect the limitHit value returned and show a + next to the number if it’s not a complete count. I think we might also want to provide fuzzier numbers for an imcomplete count, like 1+, 5+, 10+, 50+, 100+, 500+. I think that communicates the “this number might not be complete” better than 237+.
- Fix icons in the footer link panel (@taiyab can you provide a names of icons that you used in this UI, it looks like they are not standard to mdi icons that we use? not sure)
- Mobile/tablet layout version for filter UI

**Comment by camdencheek** [comment]
> Clicking a repository currently fails when the new search mode is enabled because repo: no longer takes a regex in that mode

The revert for this is [here](https://github.com/sourcegraph/sourcegraph/pull/59381)

**Comment by taiyab** [comment]
> * Fix icons in the footer link panel (@taiyab can you provide a names of icons that you used in this UI, it looks like they are not standard to mdi icons that we use? not sure)

All the icons are non-standard. I want to start introducing a new icon set to the design system, specifically [this one](https://www.streamlinehq.com/icons/core-line).

How difficult/easy would this be?

**Comment by vovakulikov** [comment]
@taiyab It's not difficult at all from a tech point of view, if we have SVG sources of these icons we're good. I can add some of them that we need here in filters panel UI. The problem is about consistency with other icon set that we currently use. But I think this part of the problem should be solved within design department

**Comment by camdencheek** [comment]
> Add count:all filter if results got their limit

I don't think this is as obvious as it sounds. The rest of our filters are subtractive: they can only return a strict subset of the results previously returned. `count` is additive. It changes the mental model of the filters panel from reducing the set of results returned by the user-entered query.

**Comment by taiyab** [comment]
@camdencheek Implementing this thing should help in alleviating some of the pain of discovering `count:all` 
<img width="602" alt="image" src="https://github.com/sourcegraph/sourcegraph/assets/5337876/52e65e3e-c17b-48fa-a0c6-50c2cd6b9e30">

However, if it's not in the LHS, we'll need to (very prominently) show that option elsewhere then. My first gut is the actual search bar itself.

**Comment by taiyab** [comment]
> @taiyab It's not difficult at all from a tech point of view, if we have SVG sources of these icons we're good. I can add some of them that we need here in filters panel UI. The problem is about consistency with other icon set that we currently use. But I think this part of the problem should be solved within design department

You can get the SVG source directly from Figma:
<img width="594" alt="image" src="https://github.com/sourcegraph/sourcegraph/assets/5337876/63f13fc2-9244-435e-9359-9f32f5fbc4cd">

**Comment by taiyab** [comment]
@vovakulikov Can we add 1-2 rounds of design polish pairing with me to the list? Or doesn't it belong there?

**Comment by vovakulikov** [comment]
@taiyab we can do it, I added it in to-do list, let's have the first round this week, review the current stage and things that you want us to fix before the Feb release.
